sql/src/lib.rs

```diff
@@ -8,18 +8,40 @@
 // CONDITIONS OF ANY KIND, either express or implied. See the License for the
 // specific language governing permissions and limitations under the License.
 
+extern crate ordered_float;
+extern crate mentat_core;
+
 use std::error::Error;
 
+use ordered_float::OrderedFloat;
+
+use mentat_core::TypedValue;
+
 pub type BuildQueryError = Box<Error + Send + Sync>;
 pub type BuildQueryResult = Result<(), BuildQueryError>;
 
+pub enum BindParamError {
+    InvalidParameterName(String),
+    BindParamCouldBeGenerated(String),
+}
+
+/// We want to accumulate values that will later be substituted into a SQL statement execution.
+/// This struct encapsulates the generated string and the _initial_ argument list.
+pub struct SQLQuery {
+    pub sql: String,
+
+    /// These will eventually perhaps be rusqlite `ToSql` instances.
+    pub args: Vec<(String, String)>,
```

Comment:
`rusqlite` uses — as does SQLite — a simple array of pairs to represent named (as opposed to positional) arguments.

I expect that if this turns into `ToSql` it'll actually be boxed `AsRef<ToSql>` or `Borrow<ToSql>` instances, or something similarly verbose.

For now, it's only keywords and strings that need this handling…